plugin/src/main/java/io/jenkins/plugins/analysis/core/steps/WarningChecksPublisher.java

```diff
@@ -0,0 +1,124 @@
+                        .build())
+                .withDetailsURL(action.getAbsoluteUrl())
+                .build();
+    }
+
+    private String extractChecksSummary(final IssuesStatistics statistics) {
+        return String.format("## %d issues in total:\n"
+                        + "- ### %d new issues\n"
+                        + "- ### %d outstanding Issues\n"
+                        + "- ### %d delta issues\n"
+                        + "- ### %d fixed issues",
+                statistics.getTotalSize(), statistics.getNewSize(), statistics.getTotalSize() - statistics.getNewSize(),
+                statistics.getDeltaSize(), statistics.getFixedSize());
+    }
+
+    private String extractChecksText(final IssuesStatistics statistics) {
+        return "## Total Issue Statistics:\n"
+                + generateSeverityText(statistics.getTotalLowSize(), statistics.getTotalNormalSize(),
+                statistics.getTotalHighSize(), statistics.getTotalErrorSize())
+                + "## New Issue Statistics:\n"
+                + generateSeverityText(statistics.getNewLowSize(), statistics.getNewNormalSize(),
+                statistics.getNewHighSize(), statistics.getNewErrorSize())
+                + "## Delta Issue Statistics:\n"
+                + generateSeverityText(statistics.getDeltaLowSize(), statistics.getDeltaNormalSize(),
+                statistics.getDeltaHighSize(), statistics.getDeltaErrorSize());
+    }
+
+    private String generateSeverityText(final int low, final int normal, final int high, final int error) {
+        return "* Error: " + error + "\n"
+                + "* High: " + high + "\n"
+                + "* Normal: " + normal + "\n"
+                + "* Low: " + low + "\n";
+    }
+
+    private ChecksConclusion extractChecksConclusion(final QualityGateStatus status) {
+        switch (status) {
+            case INACTIVE:
+            case PASSED:
+                return ChecksConclusion.SUCCESS;
+            case FAILED:
+            case WARNING:
+                return ChecksConclusion.FAILURE;
+            default:
+                throw new IllegalArgumentException("Unsupported quality gate status: " + status);
+        }
+    }
+
+    private List<ChecksAnnotation> extractChecksAnnotations(final Report issues) {
+        return issues.stream()
+                .map(issue -> new ChecksAnnotationBuilder()
+                        .withPath(issue.getFileName())
+                        .withTitle(issue.getType())
+                        .withAnnotationLevel(ChecksAnnotationLevel.WARNING)
+                        .withMessage(issue.getSeverity() + ": " + issue.getMessage())
+                        .withStartLine(issue.getLineStart())
+                        .withEndLine(issue.getLineEnd())
+                        .withStartColumn(issue.getColumnStart())
+                        .withEndColumn(issue.getColumnEnd())
+                        .withRawDetails(issue.getDescription())
```

Comment:
Then you get for CheckStyle, SpotBugs a detailed description as well:
```suggestion
                        .withRawDetails(labelProvider.getDescription(issue))
```
This description will contain HTML tags...

Reply:
the GitHub just supports markdown (or HTML tag) on the summary and description of the whole output, but not for these specific annotations.

I'll find a way to extract the plain text as the description, and left the HTML in the `raw-details` field.

Reply:
Is the `getMessage` method in `Issue` just extract the message from the HTML? Then we do not need any further changes.

Reply:
The `message` in `Issue` returns the header of an issue. The `description` returns additional information for that issue (from the parsed content). And the `description` of the `labelProvider` does return a HTML description of the bug pattern for specific tools (CheckStyle, Spotbugs, PMD), from the tools website or jar file. This part is not in the parsed content. If that part is empty the `labelProvider` does return the `description` of the issue. So none of the methods removes the HTML.